reports/aksVersions/aks-versions.sh

```diff
@@ -0,0 +1,147 @@
+# # --------------------------------------------------------------------------
+
+declare -a aksClusterInfo
+
+for sub in "${subscriptions[@]}"; do
+  
+  declare -a currentClusters
+
+  echo "Setting subscription: $sub"
+  az account set --subscription "$sub"
+  echo "Searching for AKS clusters..."
+  currentClusters+=$(az aks list --query '[*].{"Name":name,"ResourceGroup":resourceGroup, "kubernetesVersion":kubernetesVersion, "PowerState":powerState.code}' | jq .)
+  echo "Done searching!"
+
+  for cluster in $(echo "$currentClusters" | jq -c '.[]'); do
+    resourceGroup=$(echo "$cluster" | jq -r '.ResourceGroup')   
+    aksName=$(echo "$cluster" | jq -r '.Name')
+
+    echo "Checking for upgrades to: $(echo $cluster | jq -r '.Name')"
+    currentVersion=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.kubernetesVersion')
+    upgradeAvailable=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades')
+
+    if [[ $upgradeAvailable == 'null' ]]; then
+      verdict="No Update Required"
+      color_code=green
+      upgradeableVersion="None"
+    else
+      upgradesIsPreview=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades[].isPreview')
+      if [[ $upgradesIsPreview == 'null' ]]; then
+        upgradeableVersion=$(az aks get-upgrades --resource-group $resourceGroup --name $aksName | jq -r '.controlPlaneProfile.upgrades[].kubernetesVersion')
+        color_code=red
+        verdict="Update required"
+      else
+        upgradeableVersion="Preview Only"
+        color_code=orange
+        verdict="Not ready for update"
+      fi
+    fi
+
+    echo "Creating json object for: $(echo $cluster | jq -r '.Name')"
+    aksClusterInfo+=$(echo "$cluster" | jq -r '.' | jq --arg verdict "$verdict" \
+                                                      --arg color_code "$color_code" \
+                                                      --arg currentVersion "$currentVersion" \
+                                                      --arg id "$(uuidgen)" \
+                                                      --arg upgradeableVersion "$upgradeableVersion" '{id: $id, clusterName: .Name, powerState: .PowerState, currentVersion: $currentVersion, upgradeableVersion: $upgradeableVersion, colorCode: $color_code,  verdict: $verdict, resourceType: "AKS Cluster"}')  
+  done
+  unset currentClusters
+  echo "Completed subscription: $sub"
+done
+
+# # ---------------------------------------------------------------------------
+# # STEP 3:
+# # Store document
+# # ---------------------------------------------------------------------------
+
+aksClusterInfoOutput=$(echo "${aksClusterInfo[*]}" | jq -s)
+aksClusterInfoTemp=$(jq -n '$ARGS.positional' --jsonargs "${aksClusterInfoOutput[@]}")
+documents=$(echo "${aksClusterInfoTemp[*]}"| jq '.[]')
+
+store_document "$documents"
```

Comment:
Is it worth catching any potential errors here?

Reply:
The python script has try/catch error handling within it if it fails to access/delete/write to the container.
Is that what you are asking about here?

Reply:
Ah my bad, yes that covers that